### DISCUSSION TITLE
A short guide to getting DeepSpeed Working

### DISCUSSION THREAD

**Discussion by LoFiApostasy** [bmaltais/kohya_ss#2254] [General]
After a lot of frustrating attempts to get deep speed working I think i have distilled it to a short and sweet list of steps. My hope is that the accelerate script can eventually generate this file by default from the wizard.

My environment is dual 3090s with Ubuntu WSL installed.
Fresh install of Kohya including deepspeed, noting the activate here because its not in the current instructions:

````
cd kohya-ss
source ./venv/bin/activate
pip3 install deepspeed
````

Create deepspeed-config.json file and take note of the path using the following:

````
 {
    "train_batch_size": 1,
    "gradient_accumulation_steps": 4,
    "train_micro_batch_size_per_gpu": 1,
    "gradient_clipping": 1.0,
    "offload_optimizer_device": "cpu",
    "offload_param_device": "cpu",
    "zero3_init_flag": false,
    "zero_stage": 2
}
````

Then kick off configure accelerate so we can reference the file:

````
 /venv/bin/accelerate config
````

During the wizard choose YES to "Do you want to specify a json file to a DeepSpeed config?"
provide the path and file name.
example: /home/myuser/kohya-ss/deepspeed-config.json

After its working feel free to tinker with the config file using the main help for reference: https://www.deepspeed.ai/docs/config-json/

############ Search tags ############
The errors i was getting out of the box was:

````
ValueError: When using DeepSpeed `accelerate.prepare()` requires you to pass at least one of training or evaluation dataloaders or alternatively set an integer value in `train_micro_batch_size_per_gpu` in the deepspeed config fileor assign integer value to AcceleratorState().deepspeed_plugin.deepspeed_config['train_micro_batch_size_per_gpu
````

############
The file path to your deepspeed config can be updated in the kohya "default_config.yaml" located here:
/home/myuser/.cache/huggingface/accelerate/
The entry in the yaml looks like this:

````
deepspeed_config:
  deepspeed_config_file: /home/myuser/deepspeed-config.json
  zero3_init_flag: false

**Comment by Thomas2419** [comment]
I followed the exact steps, but im getting valuerror: Please specify the ZeRO optmizations config in the DeepSpeed config. Any suggestions?
